modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/model/internal/DeclarationBasedElementModelFactory.java

```diff
@@ -768,16 +799,31 @@ private void createList(ParameterListValue list,
         .withIdentifier(asIdentifier(listDsl));
 
     final MetadataType itemType = listType.getType();
-    listDsl.getGeneric(itemType)
-        .ifPresent(itemDsl -> list.getValues()
-            .forEach(value -> createListItemConfig(itemType, value, itemDsl, listConfig, listElement)));
 
+    DslElementSyntax itemDsl =
+        listDsl.getGeneric(itemType).isPresent() ? listDsl.getGeneric(itemType).get() : createSetVariableDslElement();
+    list.getValues().forEach(value -> createListItemConfig(itemType, value, itemDsl, listConfig, listElement));
     ComponentConfiguration result = listConfig.build();
 
+
     parentConfig.withNestedComponent(result);
     parentElement.containing(listElement.withConfig(result).build());
   }
 
+  private DslElementSyntax createSetVariableDslElement() {
```

Comment:
`createTransformSetVariableParameterDslElement`... if it's hackish, let's be clear about it